### ISSUE TITLE
Missing transactions in category view

### ISSUE THREAD

**Issue by Bubka** [firefly-iii/firefly-iii#1996]
**Bug description**
I am running Firefly III version 4.7.9

Some transactions are missing when I open `/categories/show/[catId]/`.
They are missing in the transactions table of the view but the transactions exist in db and their amounts are used in the category sum for the month.

See the Decembre total for `/categories/show/17/2018-12-01/2018-12-31`:
![firefly-missing-transaction-02](https://user-images.githubusercontent.com/858858/51039372-75bb8c00-15b5-11e9-86a0-810f251f8699.png)

Here are all transactions find by the search feature on Decembre 2018:
![firefly-missing-transaction-01](https://user-images.githubusercontent.com/858858/51039376-77854f80-15b5-11e9-9555-8f6e4a3d1b58.png)

If I edit one of the missing transaction it makes the transaction appearing back in the category view.

**Steps to reproduce**
Cannot reproduce

**Expected behavior**
See all transactions for the selected month and the displayed category

**Extra info**
I have converted my MySQL tables from MyISAM to InnoDB recently. Could it be related ?!

Debug information generated at 2019-01-11 15:25:02 Europe/Paris for Firefly III version **4.7.9**.

| Variable | Content |
| --- | --- |
| FF version | 4.7.9 |
| FF API version | 0.9.0 |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | , daily |
| PHP version | 7.2.12 |
| Display errors | Off |
| Session start | 2018-12-01 00:00:00 |
| Session end | 2018-12-31 00:00:00 |
| Session first | 2006-01-02 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | fpm-fcgi |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Login provider |   |
| Storage disks | local-upload  |
| Using Sandstorm? | no |
| Is Sandstorm (.env) | false |
| Is Docker (.env) | false |
| bunq uses sandbox | false |
| Trusted proxies (.env) |  |
| User agent | Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:64.0) Gecko/20100101 Firefox/64.0 |
| Loaded extensions | Core, date, libxml, openssl, pcre, sqlite3, zlib, bcmath, bz2, calendar, ctype, curl, dba, dom, hash, fileinfo, filter, ftp, gd, gettext, gmp, SPL, iconv, imagick, session, intl, json, mbstring, memcached, standard, mysqlnd, PDO, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, pspell, redis, Reflection, imap, SimpleXML, soap, sockets, sodium, mongodb, exif, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, mysqli, cgi-fcgi, ionCube Loader, Zend OPcache |

**Comment by Bubka** [comment]
erf, same issue on `/transactions/withdrawal` too

**Comment by Bubka** [comment]
Also these transactions were previously associated to a budget i removed some days ago.
They are all in my `Epicerie` category and only transactions from these are missing.

**Comment by JC5** [comment]
Mmm, very strange. It could be mysql related but I am just not sure. I'll have to test.

In the mean time, run `php artisan cache:clear` on the command line, see if it helps.

**Comment by Bubka** [comment]
Ok, bug reproduced on FF demo:
- Create a new budget `Epicerie`
- Create a new category `Epicerie`
- Create a new transaction with both budget and category `Epicerie`
- Create a new transaction with just Category `Epicerie`
- Delete the `Epicerie` budget
- Open `Epicerie`category view

The first transaction associated to the `Epicerie` budget is missing

**Comment by JC5** [comment]
Thanks for debugging this. It must have taken quite some time to get to the bottom of it.

I was able to find the cause. Ultimately, it is because the connection between the deleted budget (still in the database) and the transaction is not severed. But the database query ignores the transaction because it ignores deleted budgets. 

I've updated the demo site where this should no longer happen. It will be fixed with the next release.

Thanks again!
